activation/nipost_test.go

```diff
@@ -761,8 +761,7 @@ func TestNIPSTBuilder_PoetUnstable(t *testing.T) {
 			challenge,
 			&types.NIPostChallenge{PublishEpoch: postGenesisEpoch + 2},
 		)
-		poetErr := &PoetSvcUnstableError{}
-		require.ErrorAs(t, err, &poetErr)
+		require.ErrorIs(t, err, ErrATXChallengeExpired)
```

Comment:
It returns `ErrATXChallengeExpired` because the poet hangs indefinitely and the context used to submit expires when the poet round starts. At this point, the challenge has expired.

It may be better to handle this on the retry level. I.e. not retry and expire the challenge if it's already too late. Wdyt?

Reply:
> It may be better to handle this on the retry level. I.e. not retry and expire the challenge if it's already too late. Wdyt?

Looking at the handling of the two different errors in `activation.go`, it looks like the change to `ErrATXChallengeExpired` is correct. It will invalidate the current challenge and build a new one with the publish epoch set to current + 2.

Returning `PoetSvcUnstableError` worked before because it would retry to submit to a PoET but realize that it is already too late to submit for the round with the publish epoch set in the challenge and then return `ErrATXChallengeExpired`.

So I believe the behavior with the change in this PR is the correct one 🙂